hivemq-edge/src/frontend/src/extensions/datahub/components/controls/PolicyToolbar.tsx

```diff
@@ -0,0 +1,123 @@
+import { MdOutlineDeleteForever, MdOutlineSave } from 'react-icons/md'
+
+import { ProblemDetailsExtended } from '@/api/types/http-problem-details.ts'
+
+import useDataHubDraftStore from '@datahub/hooks/useDataHubDraftStore.ts'
+import CheckPolicyReportToast from '@datahub/components/helpers/CheckPolicyReportToast.tsx'
+import { BehaviorPolicyData, DataHubNodeData, DataPolicyData, PolicyDryRunStatus } from '@datahub/types.ts'
+import { usePolicyDryRun } from '@datahub/hooks/usePolicyDryRun.ts'
+import { useTranslation } from 'react-i18next'
+
+interface ActionPolicyCheckProps {
+  node: NodeProps<DataPolicyData | BehaviorPolicyData>
+}
+
+export const PolicyToolbar: FC<ActionPolicyCheckProps> = (props) => {
+  const { t } = useTranslation('datahub')
+  const [state, setState] = useState<PolicyDryRunStatus>(PolicyDryRunStatus.IDLE)
+  const store = useDataHubDraftStore()
+  const { nodes, onUpdateNodes } = store
+  const createToast = useToast()
+  const { fitView } = useReactFlow()
+  const { checkPolicyAsync } = usePolicyDryRun()
+
+  // only allow a single check at any given time
+  const POLICY_CHECK_ID = 'POLICY_CHECK_ID'
+
+  const TOAST_DEFAULTS: UseToastOptions = {
+    id: POLICY_CHECK_ID,
+    isClosable: true,
+    position: 'bottom-left',
+    variant: 'left-accent',
+    duration: null,
+  }
+
+  const handleCheckPolicy = () => {
+    const selectedNode = nodes.find((node) => node.id === props.node.id)
+    if (!selectedNode) return
+
+    setState(PolicyDryRunStatus.RUNNING)
+
+    if (!createToast.isActive(POLICY_CHECK_ID)) {
+      createToast.promise(checkPolicyAsync(selectedNode), {
+        success: (results) => {
+          setState(PolicyDryRunStatus.SUCCESS)
+          const failedResults = results.filter((result) => !!result.error)
+          const errors = failedResults.map((result) => result.error as ProblemDetailsExtended)
+          const status = errors.length ? 'warning' : 'success'
+
+          return {
+            title: t('workspace.dryRun.report.success.title', { context: status }),
+            status: status,
+            description: (
+              <CheckPolicyReportToast
+                errors={errors}
+                onFitView={(id) => {
+                  const errorNode = nodes.find((e) => e.id === id)
+                  if (errorNode) fitView({ nodes: [errorNode], padding: 3, duration: 800 })
+                }}
+              />
+            ),
+            ...TOAST_DEFAULTS,
+            onCloseComplete: () => {
+              nodes.forEach((node) => {
+                onUpdateNodes<DataHubNodeData>(node.id, {
+                  ...node.data,
+                  dryRunStatus: PolicyDryRunStatus.IDLE,
+                })
+              })
+            },
+          }
+        },
+        error: (e) => {
+          setState(PolicyDryRunStatus.FAILURE)
+          return { title: t('workspace.dryRun.report.error.title'), description: e.toString(), ...TOAST_DEFAULTS }
```

Comment:
issue(breaking): we may have to move the defaults up, otherwise they may overwrite the title and description later on

```suggestion
          return { ...TOAST_DEFAULTS, title: t('workspace.dryRun.report.error.title'), description: e.toString() }
```